src/objective_types/constraints.jl

```diff
@@ -179,153 +165,55 @@ function TwiceDifferentiableConstraints(c!, lx::AbstractVector, ux::AbstractVect
-            J
-        end
+   
+    function sum_constraints(_x, _λ)
+        # TODO: get rid of this allocation with DI.Cache
+        ccache_righttype = zeros(promote_type(T, eltype(_x)), nc)
+        c!(ccache_righttype, _x)
+        return sum(_λ[i] * ccache[i] for i in eachindex(_λ, ccache))
+    end
 
-        con_jac_cached = x -> begin
-            exists_cache = (cache_check[] == eltype(x))
-            if exists_cache
-                f = ref_f[]
-                return f(x)
-            else
-                jcache = zeros(eltype(x), nc)
-                out_cache = zeros(eltype(x), nc, nx)
-                cfg_cache = ForwardDiff.JacobianConfig(c!,jcache,x)
-                f = z->ForwardDiff.jacobian!(out_cache, c!, jcache, z,cfg_cache,Val{false}())
-                ref_f[] = f
-                cache_check[]= eltype(x)
-                return f(x)
-            end
-        end
+    backend = get_adtype(autodiff, chunk)
 
-    else
-        jac! = (J,x) -> con_jac!(J,x)
-   
-        #here, the cache should also include a JacobianConfig
-         con_jac_cached = x -> begin
-            exists_cache = (cache_check[] == eltype(x))
-            if exists_cache
-                f = ref_f[]
-                return f(x)
-            else
-                out_cache = zeros(eltype(x), nc, nx)
-                f = z->jac!(out_cache,x)
-                ref_f[] = f
-                cache_check[]= eltype(x)
-                return f(x)
-            end
-        end
+
+    jac_prep = DI.prepare_jacobian(c!, ccache, backend, x_example)
+    function con_jac!(_j, _x)
+        DI.jacobian!(c!, ccache, _j, jac_prep, backend, _x) 
+        return _j
     end
-    
-    hess_config_cache = ForwardDiff.JacobianConfig(typeof(con_jac_cached),lx)
-    function con_hess!(hess, x, λ)
-        ForwardDiff.jacobian!(cxxcache, con_jac_cached, x,hess_config_cache,Val{false}()) 
-        for i = 1:nc  #hot hessian loop
-            hess+=λ[i].*hi[i]
-        end
-        return hess
+
+    hess_prep = DI.prepare_hessian(sum_constraints, backend, x_example, DI.Constant(λ_example))
+    function con_hess!(_h, _x, _λ)
+        DI.hessian!(sum_constraints, _h, hess_prep, backend, _x, DI.Constant(_λ)) 
+        return _h
     end
     
-    return TwiceDifferentiableConstraints(c!, jac!, con_hess!, bounds) 
+    return TwiceDifferentiableConstraints(c!, con_jac!, con_hess!, bounds) 
 end
 
-
-function twicediff_constraints_finite(c!,lx,ux,lc,uc,fdtype,con_jac! = nothing)
-    bounds = ConstraintBounds(lx, ux, lc, uc)
-    T = eltype(bounds)
-    nx = length(lx)
-    nc = length(lc)
-    xcache = zeros(T, nx)
-    ccache = zeros(T, nc)
+function TwiceDifferentiableConstraints(c!, con_jac!,lx::AbstractVector, ux::AbstractVector,
+    lc::AbstractVector, uc::AbstractVector,
+    autodiff::Symbol = :central,
+    chunk::ForwardDiff.Chunk = checked_chunk(lx))
+    # TODO: is con_jac! still useful? we ignore it here
```

Comment:
What should we do about this? The new version of the code directly computes the Hessian of the sum of constraints